app/screens/redeem-lnurl-withdrawal-screen/redeem-bitcoin-result-screen.tsx

```diff
@@ -0,0 +1,395 @@
+          } = data
+
+          if (errors && errors.length !== 0) {
+            console.error(errors, "error with lnInvoiceCreate")
+            setErr(LL.ReceiveBitcoinScreen.error())
+            return
+          }
+          invoice && setInvoice(invoice)
+        }
+      } catch (err) {
+        console.error(err, "error with AddInvoice")
+        setErr(`${err}`)
+        throw err
+      }
+    },
+    [lnInvoiceCreate, lnUsdInvoiceCreate, walletId, type, satAmountInUsd, LL],
+  )
+
+  const submitLNURLWithdrawRequest = useCallback(
+    async (generatedInvoice) => {
+      const url = `${callback}${callback.includes("?") ? "&" : "?"}k1=${k1}&pr=${
+        generatedInvoice.paymentRequest
+      }`
+
+      const result = await fetch(url)
+
+      if (result.ok) {
+        const lnurlResponse = await result.json()
+        if (lnurlResponse?.status?.toLowerCase() === "ok") {
+          setProcessingPayment(true)
+        } else {
+          console.error(lnurlResponse, "error with redeeming")
+          setErr(LL.RedeemBitcoinScreen.redeemingError())
+          if (lnurlResponse?.reason) {
+            setLnServiceErrorReason(lnurlResponse.reason)
+          }
+        }
+      } else {
+        console.error(result.text(), "error with submitting withdrawalRequest")
+        setErr(LL.RedeemBitcoinScreen.submissionError())
+      }
+    },
+    [callback, LL, k1],
+  )
+
+  useEffect((): void | (() => void) => {
+    if (withdrawalInvoice) {
+      submitLNURLWithdrawRequest(withdrawalInvoice)
+    } else {
+      createWithdrawRequestInvoice({ satAmount, memo })
+    }
+  }, [
+    withdrawalInvoice,
+    memo,
+    satAmount,
+    createWithdrawRequestInvoice,
+    submitLNURLWithdrawRequest,
+  ])
+
+  const renderSuccessView = useMemo(() => {
+    if (invoicePaid) {
+      return (
+        <View style={styles.container}>
+          <View {...testProps("Success Icon")} style={styles.container}>
+            <GaloyIcon name={"payment-success"} size={128} />
+          </View>
+        </View>
+      )
+    }
+    return null
+  }, [invoicePaid])
+
+  const renderErrorView = useMemo(() => {
+    if (err !== "") {
+      return (
+        <View style={styles.container}>
+          <View style={styles.errorContainer}>
+            {lnServiceErrorReason && (
+              <Text style={styles.errorText} selectable>
+                {lnServiceErrorReason}
+              </Text>
+            )}
+            <Text style={styles.errorText} selectable>
+              {err}
+            </Text>
+          </View>
+        </View>
+      )
+    }
+
+    return null
+  }, [err, lnServiceErrorReason])
+
+  const renderActivityStatusView = useMemo(() => {
+    if (err === "" && !invoicePaid) {
+      return (
+        <View style={styles.container}>
+          <View>
+            <ActivityIndicator
+              size="large"
+              color={processingPayment ? palette.green : palette.blue}
```

Comment:
this should only be green when invoice is paid... with the current behavior is when lnurl withdraw service return a response

Reply:
that check is in line 224

Reply:
The check mark doesnt come until it is paid in line 256

Reply:
Yes, this is just to change the loading indicator colour when we get an okay response from the ln service processing the withdrawal.